arrows/vxl/convert_image.cxx

```diff
@@ -0,0 +1,483 @@
+        return image_data;                                             \
+      }                                                                \
+                                                                       \
+      vil_image_view< ipix_t > input;                                  \
+                                                                       \
+      if( d->random_greyscale > 0.0 )                                  \
+      {                                                                \
+        vil_image_view< ipix_t > tmp = view;                           \
+        random_grey_conversion( tmp, input, d->random_greyscale );     \
+      }                                                                \
+      else if( d->single_channel )                                     \
+      {                                                                \
+        vil_image_view< ipix_t > tmp = view;                           \
+        combine_channels( tmp, input );                                \
+      }                                                                \
+      else                                                             \
+      {                                                                \
+        input = view;                                                  \
+      }                                                                \
+      HANDLE_OUTPUT_CASE("bool", VIL_PIXEL_FORMAT_BOOL);               \
+      HANDLE_OUTPUT_CASE("byte", VIL_PIXEL_FORMAT_BYTE);               \
+      HANDLE_OUTPUT_CASE("sbyte", VIL_PIXEL_FORMAT_SBYTE);             \
+      HANDLE_OUTPUT_CASE("uint16", VIL_PIXEL_FORMAT_UINT_16);          \
+      HANDLE_OUTPUT_CASE("int16", VIL_PIXEL_FORMAT_INT_16);            \
+      HANDLE_OUTPUT_CASE("uint32", VIL_PIXEL_FORMAT_UINT_32);          \
+      HANDLE_OUTPUT_CASE("int32", VIL_PIXEL_FORMAT_INT_32);            \
+      HANDLE_OUTPUT_CASE("uint64", VIL_PIXEL_FORMAT_UINT_64);          \
+      HANDLE_OUTPUT_CASE("int64", VIL_PIXEL_FORMAT_INT_64);            \
+      HANDLE_OUTPUT_CASE("float", VIL_PIXEL_FORMAT_FLOAT);             \
+      HANDLE_OUTPUT_CASE("double", VIL_PIXEL_FORMAT_DOUBLE);           \
+    }                                                                  \
+    break;                                                             \
+
+  switch( view->pixel_format() )
+  {
+    HANDLE_INPUT_CASE(VIL_PIXEL_FORMAT_BOOL);
+    HANDLE_INPUT_CASE(VIL_PIXEL_FORMAT_BYTE);
+    HANDLE_INPUT_CASE(VIL_PIXEL_FORMAT_SBYTE);
+    HANDLE_INPUT_CASE(VIL_PIXEL_FORMAT_UINT_16);
+    HANDLE_INPUT_CASE(VIL_PIXEL_FORMAT_INT_16);
+    HANDLE_INPUT_CASE(VIL_PIXEL_FORMAT_UINT_32);
+    HANDLE_INPUT_CASE(VIL_PIXEL_FORMAT_INT_32);
+    HANDLE_INPUT_CASE(VIL_PIXEL_FORMAT_UINT_64);
+    HANDLE_INPUT_CASE(VIL_PIXEL_FORMAT_INT_64);
+    HANDLE_INPUT_CASE(VIL_PIXEL_FORMAT_FLOAT);
+    HANDLE_INPUT_CASE(VIL_PIXEL_FORMAT_DOUBLE);
```

Comment:
I'd recommend that we trim down this set of types to the ones we actually plan to support.  This 11x11 set of type expansions is going to create a lot of code bloat.  There will be 121 different copies of many of the functions in this files, most of which will never be used.  We could probably limit this to common types.  We could also refactor the code a bit to limit code bloat by templating only when really necessary.

These changes are not necessary to merge.  I'm just pointing out an area of possible improvement.